File: 上學期期末考/[2017AF-B]刀槍不入、神功護體.cpp
#include <bits/stdc++.h>
using namespace std;

#define miyuki_is_my_wife_and_sister ios::sync_with_stdio(false), cin.tie(nullptr)

int main() {
    miyuki_is_my_wife_and_sister;

    /*
        *
        *
        *
        *
        *
        *
        *
        *
11111111111111111
 222222222222222
  3333333333333
   44444444444
    555555555
     6666666
      77777
       888
        9
    */

    int x{};
    cin >> x;

    if (x & 1) {
        for (int i = 1; i < x; ++i) {
            for (int j = 0; j < x - 1; ++j) {
                cout << " ";
            }
            cout << "*\n";
        }

        for (int i = 1; i <= x; ++i) {
            for (int j = 1; j < i; ++j) {
                cout << " ";
            }
            for (int j = 1; j < (x - i + 1) * 2; ++j) {
                cout << i;
            }
            cout << '\n';
        }
    } else {
        cout<<"What the fuck do you think you're doing?\n";
    }

    return 0;
}

File: 上學期期末考/[2017AF-B]刀槍不入、神功護體.py
def main():
    x = int(input())
    
    if x % 2:
        for i in range(1, x):
            for j in range(0, x - 1):
                print(" ", end='')
            print("*")
        
        for i in range(1, x + 1):
            for j in range(1, i):
                print(" ", end='')
            
            for j in range(1, (x - i + 1) * 2):
                print(i, end='')
            print()
    else:
        print("What the fuck do you think you're doing?")

if __name__ == "__main__":
    # miyuki is my wife and sister

    main()
